courseware/api.py

```diff
@@ -564,12 +582,24 @@ def retry_failed_edx_enrollments():
         course_run = enrollment.run
         try:
             enroll_in_edx_course_runs(user, [course_run])
+        except EdxApiEnrollErrorException as exc:
```

Comment:
I did a little digging about another approach to make it less generic e.g. to check for the error message from edX in case a user is already enrolled and we try to enroll him again here, So what I was thinking about a similar logic like [this](https://github.com/mitodl/mitxpro/blob/master/courseware/api.py#L523-L540) where we check the error message to see what actually went wrong. 
But during my testing, the edX doesn't send an error message if a user is enrolled and we try to enroll again,  instead, it treats the request as a success. 

This is just for info and I don't think it needs to be addressed.